scipy/stats/_sensitivity_analysis.py

```diff
@@ -446,7 +456,7 @@ def sobol_indices(
     ...     random_state=rng
     ... )
     >>> indices.first_order
-    array([[0.31499073, 0.44011056, 0.00167054]])
+    array([[3.14996175e-01, 4.40110556e-01, 2.17329918e-04]])
```

Comment:
Do we want to respect user shapes here? It seems that:
- If the user's function is scalar input - scalar output -> 0d.
- vector input - scalar output -> 1d
- scalar input - vector output -> 1d
- vector input - vector output -> 2d
(by vector, I mean multiple variables)

I suspect that we can reconsider using `atleast_2d` everwhere; all that might just be unnecessary. Then an alternative is to just compute what the output shape should be and reshape just before returning.

Reply:
I will have a look 👍

Reply:
Ok here are my thoughts after having a look at the problem and trying to write some tests.

I am not sure we care about the input because we will always pass a NumPy 2D array to the function. We have control over that as we are the consumer of `func`. I think this should be a hard requirement otherwise we might need to add something like `vectorized` parameter which I am not a fan of. And adds lot of complexity for little added value IMO.

For the output, the function is either returning a single value or multiple ones. And you would do an analysis if you have multiple inputs. Hence the minimal output of `indices.first_order` would be `(2,)`. We could return a list or flat array in that case (`(d,)`). Otherwise we have a 2D array.

I propose to do the following: return a flat array if `func` returns a single output, otherwise return a 2D array.

I will prepare this and we can talk this through today before I push.